orttraining/orttraining/python/training/ortmodule/options.py

```diff
@@ -0,0 +1,338 @@
+            The output directory of the onnx models by default is set to the current working directory.
+            To change the output directory, the environment variable "ORTMODULE_SAVE_ONNX_PATH" can be
+            set to the destination directory path.
+        onnx_prefix (:obj:`str`, optional): Name prefix to the ORTModule ONNX models saved file names.
+            Must be provided if save_onnx is True
+
+    Raises:
+        OSError: If save_onnx is True and output directory is not writable.
+        TypeError: If save_onnx is True and name_prefix is not a valid string. Or if
+            log_level is not an instance of LogLevel.
+        ValueError: If save_onnx is True and name_prefix is an empty string.
+
+    """
+
+    def __init__(self, log_level=LogLevel.WARNING, save_onnx=False, onnx_prefix="", save_path="", config=None):
+        self.log_level = log_level
+        self.save_onnx = save_onnx
+        self.onnx_prefix = onnx_prefix
+
+        self._save_onnx_models = _SaveOnnxOptions(self.save_onnx, self.onnx_prefix, save_path)
+        self._logging = _LoggingOptions(self.log_level)
+
+    @property
+    def save_onnx_models(self):
+        """Accessor for the ONNX saving configuration."""
+
+        return self._save_onnx_models
+
+    @property
+    def logging(self):
+        """Accessor for the logging configuration."""
+
+        return self._logging
+
+
+class _SkipCheck(IntFlag):
+    """Enumeration to specify which checks should be skipped, allowing faster execution"""
+
+    SKIP_CHECK_DISABLED = 1
+    SKIP_CHECK_DEVICE = 2
+    SKIP_CHECK_BUILD_GRADIENT = 4
+    SKIP_CHECK_EXECUTION_AGENT = 8
+
+    def is_set(self, check):
+        """Check whether `check` is set on the `_SkipCheck instance
+
+        SKIP_CHECK_DISABLED implies the check will return False
+        """
+
+        return not _SkipCheck.is_disabled(self) and check in self
+
+    def is_disabled(self):
+        """Check whether `_SkipCheck.SKIP_CHECK_DISABLED is set on the `_SkipCheck instance"""
+
+        return _SkipCheck.SKIP_CHECK_DISABLED in self
+
+
+class RuntimeOptions:
+    """Configurable runtime options for ORTModule."""
+
+    def __init__(self, logger: Logger):
+        """Constructor for Options.
+
+        Initially set all the options to their default values, then override them with the values
+        from the environment variables.
+        """
+        self._logger = logger
+
+        self.onnx_opset_version = ortmodule.ONNX_OPSET_VERSION
+        self.conv_algo_search = "HEURISTIC"
+
+        # Configuration for cast optimization.
+        # Specify cast propagation strategy. Currently, three strategies are available:
+        #  NONE, INSERT-AND-REDUCE and FLOOD-FILL
+        # The default is FLOOD_FILL, expand FP16 computation regions in the graph using
+        # allowed opcodes for the given level.
+        self._propagate_cast_ops_strategy = C.PropagateCastOpsStrategy.FLOOD_FILL
```

Comment:
```suggestion
        self.propagate_cast_ops_strategy = C.PropagateCastOpsStrategy.FLOOD_FILL
```
Several options in the class are starting with `_` which in python means "don't use outside this class".